corehq/apps/custom_data_fields/edit_model.py

```diff
@@ -287,6 +287,16 @@ def get_definition(self):
     def get_profiles(self):
         return self.get_definition().get_profiles()
 
+    @classmethod
+    @memoized
+    def get_definition_for_domain(cls, domain):
+        return CustomDataFieldsDefinition.get_or_create(domain, cls.field_type)
+
+    @classmethod
+    @memoized
+    def get_user_accessible_profiles(cls, domain, *args):
+        return cls.get_definition_for_domain(domain).get_profiles()
```

Comment:
Ah I did that before I learned that profiles are specific to only `UserFieldsView`. With this [check](https://github.com/dimagi/commcare-hq/blob/dad9a4cf971361d51d67925b55ce61d6f6bb507e/corehq/apps/custom_data_fields/edit_entity.py#L143-L147), that function is no longer needed (deleted bbc841653d9)

As a side note, I don't particularly like how profiles are implemented within [init_form](https://github.com/dimagi/commcare-hq/blob/dad9a4cf971361d51d67925b55ce61d6f6bb507e/corehq/apps/custom_data_fields/edit_entity.py#L139) because it depends on `field_view` being a UserFieldsView which conflicts with the original polymorphic design. I thought about it for a while but couldn't figure out a good way to refactor it. But added the conditional check at least to make it explicitly known.